indexing-service/src/test/java/org/apache/druid/indexing/common/task/IndexTaskTest.java

```diff
@@ -1783,6 +1788,36 @@ public void testIndexTaskWithSingleDimPartitionsSpecThrowingException() throws E
     task.isReady(createActionClient(task));
   }
 
+  @Test
+  public void testShardSpecSelectionWithNullPartitionDimension()
+  {
+    ShardSpec spec1 = new HashBasedNumberedShardSpec(0, 2, null, jsonMapper);
+    ShardSpec spec2 = new HashBasedNumberedShardSpec(1, 2, null, jsonMapper);
+
+    Map<Interval, List<ShardSpec>> shardSpecMap = new HashMap<>();
+    shardSpecMap.put(Intervals.of("2014-01-01T00:00:00.000Z/2014-01-02T00:00:00.000Z"), Lists.newArrayList(spec1, spec2));
+
+    IndexTask.ShardSpecs shardSpecs = new IndexTask.ShardSpecs(shardSpecMap, Granularity.fromString("HOUR"));
+    String visitor_id = "visitor_id";
+    String client_type = "client_type";
```

Comment:
The Java convention is the camel case. Please rename these variables to `visitorId` and `clientType`.

Reply:
Fixed